include/mbedtls/check_config_psa.h

```diff
@@ -0,0 +1,87 @@
+/**
+ * \file check_config_psa.h
+ *
+ * \brief Consistency checks for PSA configuration options
+ */
+/*
+ *  Copyright The Mbed TLS Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * It is recommended to include this file from your config_psa.h
+ * in order to catch dependency issues early.
+ */
+
+#ifndef MBEDTLS_CHECK_CONFIG_PSA_H
+#define MBEDTLS_CHECK_CONFIG_PSA_H
+
+#if defined(MBEDTLS_PSA_BUILTIN_ALG_DETERMINISTIC_ECDSA) && \
```

Comment:
My first comment was about MBEDTLS_PSA_BUILTIN_xxx macros but as I said above eventually I think the checks should be on PSA_WANT_xxx macros. It is to check the consistency of the crypto_config.h file thus this file should probably be named check_crypto_config.h and be located in include/psa.

Because of the policy "if A needs B then selecting A selects B" for the PSA_WANT_xxx macros check_crypto_config.h should be much smaller than check_config.h but we need one for at least A => B or C.